src/common.ts

```diff
@@ -0,0 +1,56 @@
+import { addMinutes, format, set } from 'date-fns';
+import type { z } from 'zod';
+
+export function formatDate(date: Date): string {
+  return format(date, 'yyyy-MM-dd');
+}
+
+/**
+ * @param date ISO 8601 date string
+ * @returns formatted matching Abax API date format
+ */
+export function formatDateTime(date: Date): string {
+  return date.toISOString();
+}
+
+export function startOfTheNextMinute(fromDate?: Date): Date {
+  const now = fromDate ?? new Date();
+
+  const zeroSeconds = set(now, {
+    seconds: 0,
+    milliseconds: 0,
+  });
+
+  return addMinutes(zeroSeconds, 1);
+}
+
+/**
+ *
+ * @param data search parameters
+ * @returns the parameters formatted as URLSearchParams
+ *
+ * @example  makeSearchParams({ foo: 'bar', baz: 42 }) => 'foo=bar&baz=42'
+ *
+ * Create search parameters from a flat object.
+ */
+export function makeSearchParams(
+  data: Record<string, unknown>,
+): URLSearchParams {
+  const params = new URLSearchParams();
+
+  Object.entries(data).forEach(([key, value]) => {
+    if (value) {
+      params.append(key, String(value));
+    }
+  });
+
+  return params;
+}
```

Comment:
```suggestion
/** Creates search params from a one-dimensional object. Supports array values. */
export function makeSearchParams(
  data: Record<string, unknown>,
): URLSearchParams {
  const params = new URLSearchParams();
  Object.entries(data).forEach(([key, value]) => {
    if (value) {
      if (Object.prototype.toString.call(value) === '[object Array]') {
        Object.values(value).forEach(v => {
          params.append(key, makeStringFromSeachParam(v));
        });
      } else if (value instanceof Date) {
        params.append(key, makeStringFromSeachParam(value));
      } else {
        params.append(key, makeStringFromSeachParam(value));
      }
    }
  });

  return params;
}
```

I think supporting array values is useful

Reply:
```
      } else if (value instanceof Date) {
        params.append(key, makeStringFromSeachParam(value));
      } else {
        params.append(key, makeStringFromSeachParam(value));
      }
```
I don't understand the difference between these ifs?

Reply:
Heh, you're right, something got lost in refactoring in that snippet.